lib/cloud_controller/opi/apps_client.rb

```diff
@@ -0,0 +1,134 @@
+require 'uri'
+require 'httpclient'
+require 'multi_json'
+require_relative '../../vcap/vars_builder'
+require 'json'
+require 'ostruct'
+
+module OPI
+  class Client
+    def initialize(opi_url)
+      @client = HTTPClient.new(base_url: URI(opi_url))
+    end
+
+    def desire_app(process)
+      process_guid = process_guid(process)
+      path = "/apps/#{process_guid}"
+      @client.put(path, body: desire_body(process))
+    end
+
+    def fetch_scheduling_infos
+      path = '/apps'
+
+      resp = @client.get(path)
+      resp_json = JSON.parse(resp.body)
+      resp_json['desired_lrp_scheduling_infos'].map { |h| recursive_ostruct(h) }
+    end
+
+    def update_app(process, _)
+      path = "/apps/#{process.guid}"
+
+      response = @client.post(path, body: update_body(process))
+      if response.status_code != 200
+        response_json = recursive_ostruct(JSON.parse(response.body))
+        raise CloudController::Errors::ApiError.new_from_details('RunnerError', response_json.error.message)
+      end
+      response
+    end
+
+    def get_app(process)
+      path = "/apps/#{process.guid}"
+
+      response = @client.get(path)
+      if response.status_code == 404
+        return nil
+      end
+
+      desired_lrp_response = recursive_ostruct(JSON.parse(response.body))
+      desired_lrp_response.desired_lrp
+    end
+
+    def stop_app(process_guid)
+      path = "/apps/#{process_guid}/stop"
+      @client.put(path)
+    end
+
+    def bump_freshness; end
```

Comment:
Yes, it is. It is not yet implemented by OPI, but since it's always called by the [ProcessesSync](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/lib/cloud_controller/diego/processes_sync.rb#L67) (as long as there are no Errors), we have to provide it.

Thanks,
Georgi && @suhlig